cmd/packaging/windows-msi.go

```diff
@@ -24,26 +31,93 @@ var WindowsMsiTask = &packagingTask{
 	flutterBuildOutputDirectory: "build",
 	packagingFunction: func(tmpPath, applicationName, strippedApplicationName, packageName, executableName, version, release string) (string, error) {
 		outputFileName := fmt.Sprintf("%s %s.msi", applicationName, version)
-		cmdConvert := exec.Command("convert", "-resize", "x16", "build/assets/icon.png", "build/assets/icon.ico")
-		cmdConvert.Dir = tmpPath
-		cmdConvert.Stdout = os.Stdout
-		cmdConvert.Stderr = os.Stderr
-		err := cmdConvert.Run()
+		f, err := os.Open(filepath.Join(tmpPath, "build", "assets", "icon.png"))
+		if err != nil {
+			return "", err
+		}
+		pngImage, err := png.Decode(f)
+		if err != nil {
+			return "", err
+		}
+		// We can't defer it, because windows reports that the file is used by another program
+		err = f.Close()
+		if err != nil {
+			return "", err
+		}
+		icoImage, err := os.Create(filepath.Join(tmpPath, "build", "assets", "icon.ico"))
+		if err != nil {
+			return "", err
+		}
+		err = ico.Encode(icoImage, pngImage)
 		if err != nil {
 			return "", err
 		}
-		cmdWixl := exec.Command("wixl", "-v", fmt.Sprintf("%s.wxs", packageName), "-o", outputFileName)
-		cmdWixl.Dir = tmpPath
-		cmdWixl.Stdout = os.Stdout
-		cmdWixl.Stderr = os.Stderr
-		err = cmdWixl.Run()
+		// We can't defer it, because windows reports that the file is used by another program
+		err = icoImage.Close()
 		if err != nil {
 			return "", err
 		}
+		if runtime.GOOS == "windows" {
```

Comment:
I prefer switches for OS-specific code like this because it makes it more clear which OS's are supported here and which aren't. The `default:` case could `panic("should be unreachable")` since we check for host-OS compatibility before running this function. If somehow the host-OS compatibility check is bugged, a panic is better than continuing as if nothing is wrong.

It's also relevant to developers reading this code to know that darwin support is missing here. With the if/elseif, it looks as if we can continue without doing anything specific for darwin host-OS. (or freeBSD... or dragonfly...)